lib/metrics.py

```diff
@@ -82,49 +87,54 @@ def __init__(self, interval, m2ee):
             logger.info("Metrics are logged to stdout.")
             self.emitter = LoggingEmitter()
 
-    def emit(self, stats):
+    @staticmethod
+    def _set_stats_info(stats):
         stats["version"] = "1.0"
         stats["timestamp"] = datetime.datetime.now().isoformat()
         stats["instance_index"] = os.getenv("CF_INSTANCE_INDEX", 0)
+        return stats
+
+    def emit(self, stats):
         self.emitter.emit(stats)
 
+    @abstractmethod
```

Comment:
I would make this an `abstractproperty` instead of an abstract method, and make the subclass implementations properties too.

Reply:
Cloud you elaborate why?

Reply:
https://stackoverflow.com/questions/20506612/python-property-vs-method-when-no-access-to-attribute-is-needed

tl;dr returning simple values that are essentially static/constant for an instance of the class, it's just more elegant.